File: sample-aac/src/main/java/sky/example/MainBiz.java
package sky.example;

import android.os.Bundle;

import sk.SKBiz;
import sk.SKHelper;
import sk.SKToast;
import sky.OpenMethod;
import sky.example.di.TestID;
import sky.example.http.model.Model;

/**
 * @author sky
 * @version 1.0 on 2018-04-28 下午3:36
 * @see MainBiz
 */
public class MainBiz extends SKBiz {

	@Override public void initBiz(Bundle bundle) {}

	@OpenMethod(TestID.A) public void a() {
		SKHelper.toast().show("我被执行了a()");
	}

	@OpenMethod(TestID.B) public void a(int a, Model model) {
		SKHelper.toast().show("我被执行了a()" + a + ":" + model);
	}

	@OpenMethod(TestID.C) public int b() {
		return 222;
	}

	@OpenMethod(TestID.D) public Model c() {
		return new Model();
	}
}

File: sample-kotlin/src/main/java/sky/sample/MainBiz.kt
package sky.sample

import android.util.Log
import sky.Background
import sky.BackgroundType
import sky.core.SKYBiz
import sky.core.SKYHelper

/**
 * @author sky
 * @version 1.0 on 2017-12-06 下午4:36
 * @see MainBiz
 */
open class MainBiz : SKYBiz<MainActivity>() {


    @Background(BackgroundType.WORK) open fun a() {

        Thread.sleep(3000)

        SKYHelper.toast().show("弹框")

    }

    open fun b(name: String, number: Int) {
        Log.i("mainbiz", "名字 ${name} 年龄 ${number}")
    }

}
